conftest.py

```diff
@@ -15,9 +15,13 @@
 
 
 @pytest.hookimpl(hookwrapper=True)
-def pytest_runtest_teardown(item, nextitem):
-    """Called after pytest_runtest_call."""
-    outcome = yield
-    rs = es().es.search(index="apm-*")
-    with open(f'/app/tests/results/data-{item.name}.json', 'w') as outFile:
-        json.dump(rs, outFile, sort_keys=True, indent=4, ensure_ascii=False)
+def pytest_runtest_logreport(report):
+    yield
+    if report.when == "call" and report.failed:
+        rs = es().es.search(index="apm-*", size=1000)
```

Comment:
I have put 10000 in a previous PR, 1000 could not grab all the data

Reply:
Do we really need all of the documents at that point though?  I expect 10000 could cause some problems on these wimpy machines

Reply:
if you want to check the data stored to see what could happen, you'd need all the data, for example, if you want to check why there are more documents than expected, I saw this error many times and without the data, you cannot figure out what it is the error.

Reply:
I understand why you'd want that but I"m concerned that it's going to cause problems for these tiny instances.  The tests do cheap count queries whereas this is pulling back the bodies for 1000s of documents - at that point I'd expect to run this test locally and run aggregation queries to figure out what is wrong, not process this debugging file.

Reply:
you will run this when you are wrapping up the test, so you would not affect the test, I have checked it locally and the bigger file is about 15MB